### ISSUE TITLE
[BUG] fltk-sys build script fails when cross compiling to x86_64-pc-windows-gnu

### ISSUE THREAD

**Issue by EthanWeegee** [fltk-rs/fltk-rs#1332]
## Describe the bug
Compiling to x86_64-pc-windows-gnu gives an error while compiling fluid. It seems to be complaining about finding a git repository, which seems really strange. I don't know what is causing this.

## To Reproduce
I'm just running `cargo build --release --target=x86_64-pc-windows-gnu` inside of [this repository](https://github.com/EthanWeegee/simple-nus3audio-gui).

## Screenshots
Here's the whole output:

```
$ cargo build --release --target=x86_64-pc-windows-gnu
   Compiling fltk-sys v1.2.20
error: failed to run custom build command for `fltk-sys v1.2.20`

Caused by:
  process didn't exit successfully: `/home/ethan/src/projects/simple-nus3audio-gui/target/release/build/fltk-sys-14d6fdba50a175d6/build-script-main` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=build/android.rs
  cargo:rerun-if-changed=build/bundled.rs
  cargo:rerun-if-changed=build/link.rs
  cargo:rerun-if-changed=build/main.rs
  cargo:rerun-if-changed=build/source.rs
  cargo:rerun-if-changed=build/utils.rs
  cargo:rerun-if-env-changed=CC
  cargo:rerun-if-env-changed=CXX
  cargo:rerun-if-changed=cfltk/CMakeLists.txt
  cargo:rerun-if-changed=cfltk/include/cfl.h
  cargo:rerun-if-changed=cfltk/include/cfl_widget.h
  cargo:rerun-if-changed=cfltk/include/cfl_group.h
  cargo:rerun-if-changed=cfltk/include/cfl_input.h
  cargo:rerun-if-changed=cfltk/include/cfl_window.h
  cargo:rerun-if-changed=cfltk/include/cfl_button.h
  cargo:rerun-if-changed=cfltk/include/cfl_box.h
  cargo:rerun-if-changed=cfltk/include/cfl_menu.h
  cargo:rerun-if-changed=cfltk/include/cfl_dialog.h
  cargo:rerun-if-changed=cfltk/include/cfl_valuator.h
  cargo:rerun-if-changed=cfltk/include/cfl_browser.h
  cargo:rerun-if-changed=cfltk/include/cfl_misc.h
  cargo:rerun-if-changed=cfltk/include/cfl_text.h
  cargo:rerun-if-changed=cfltk/include/cfl_image.h
  cargo:rerun-if-changed=cfltk/include/cfl_draw.h
  cargo:rerun-if-changed=cfltk/include/cfl_table.h
  cargo:rerun-if-changed=cfltk/include/cfl_surface.h
  cargo:rerun-if-changed=cfltk/include/cfl_printer.h
  cargo:rerun-if-changed=cfltk/include/cfl_utils.h
  cargo:rerun-if-changed=cfltk/include/cfl_macros.h
  cargo:rerun-if-changed=cfltk/include/cfl_lock.h
  cargo:rerun-if-changed=cfltk/src/cfl_lock.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_new.cpp
  cargo:rerun-if-changed=cfltk/src/cfl.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_widget.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_group.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_window.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_button.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_box.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_menu.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_dialog.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_valuator.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_browser.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_misc.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_text.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_image.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_input.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_draw.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_table.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_tree.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_surface.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_printer.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_font.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_utils.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_nswindow.m
  cargo:rerun-if-changed=cfltk/FL_FLex/FL_Flex.H
  cargo:rerun-if-changed=cfltk/FL_FLex/FL_Flex.cpp
  cargo:rerun-if-changed=cfltk/src/cfl_vec.hpp
  CMAKE_TOOLCHAIN_FILE_x86_64-pc-windows-gnu = None
  CMAKE_TOOLCHAIN_FILE_x86_64_pc_windows_gnu = None
  TARGET_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-pc-windows-gnu = None
  CMAKE_GENERATOR_x86_64_pc_windows_gnu = None
  TARGET_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-pc-windows-gnu = None
  CMAKE_PREFIX_PATH_x86_64_pc_windows_gnu = None
  TARGET_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-pc-windows-gnu = None
  CMAKE_x86_64_pc_windows_gnu = None
  TARGET_CMAKE = None
  CMAKE = None
  running: "cmake" "/home/ethan/.cargo/registry/src/github.com-1ecc6299db9ec823/fltk-sys-1.2.20/cfltk" "-DCMAKE_SYSTEM_NAME=Windows" "-DCMAKE_RC_COMPILER=/usr/bin/x86_64-w64-mingw32-windres" "-DOPTION_USE_SYSTEM_LIBPNG=OFF" "-DOPTION_USE_SYSTEM_LIBJPEG=OFF" "-DOPTION_USE_SYSTEM_ZLIB=OFF" "-DCFLTK_LINK_IMAGES=ON" "-DOpenGL_GL_PREFERENCE=GLVND" "-DOPTION_USE_GL=OFF" "-DCFLTK_USE_OPENGL=OFF" "-DCFLTK_SINGLE_THREADED=OFF" "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON" "-DCFLTK_CARGO_BUILD=ON" "-DFLTK_BUILD_EXAMPLES=OFF" "-DFLTK_BUILD_TEST=OFF" "-DOPTION_LARGE_FILE=ON" "-DOPTION_USE_THREADS=ON" "-DOPTION_BUILD_HTML_DOCUMENTATION=OFF" "-DOPTION_BUILD_PDF_DOCUMENTATION=OFF" "-DCMAKE_INSTALL_PREFIX=/home/ethan/src/projects/simple-nus3audio-gui/target/x86_64-pc-windows-gnu/release/build/fltk-sys-cff6bce13894592f/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -m64" "-DCMAKE_C_COMPILER=/usr/bin/x86_64-w64-mingw32-gcc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -m64" "-DCMAKE_CXX_COMPILER=/usr/bin/x86_64-w64-mingw32-g++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -m64" "-DCMAKE_ASM_COMPILER=/usr/bin/x86_64-w64-mingw32-gcc" "-DCMAKE_BUILD_TYPE=Release"
  -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
  -- Configuration Summary for FLTK 1.4.0 generated by CMake 3.25.1 --

  -- The following OPTIONAL packages have not been found:

   * Doxygen

  -- Static libraries will be built in /home/ethan/src/projects/simple-nus3audio-gui/target/x86_64-pc-windows-gnu/release/build/fltk-sys-cff6bce13894592f/out/build/fltk/lib
  -- Shared libraries will not be built (set OPTION_BUILD_SHARED_LIBS=ON to build)
  -- Example programs will not be built (set FLTK_BUILD_EXAMPLES=ON to build)
  -- End of Configuration Summary --

  -- Configuring done
  -- Generating done
  -- Build files have been written to: /home/ethan/src/projects/simple-nus3audio-gui/target/x86_64-pc-windows-gnu/release/build/fltk-sys-cff6bce13894592f/out/build
  running: "cmake" "--build" "." "--target" "install" "--config" "Release" "--parallel" "16"
  [  5%] Built target fltk_z
  [ 19%] Built target fltk_jpeg
  [ 26%] Built target cfltk
  [ 32%] Built target fltk_png
  [ 86%] Built target fltk
  [ 88%] Built target fltk_forms
  [ 92%] Built target fltk_images
  [ 92%] Building CXX object fltk/fluid/CMakeFiles/fluid.dir/ExternalCodeEditor_WIN32.cxx.obj

  --- stderr
  fatal: not a git repository (or any of the parent directories): .git
  error: patch failed: src/CMakeLists.txt:303
  error: src/CMakeLists.txt: patch does not apply
  error: patch failed: src/Fl_win32.cxx:603
  error: src/Fl_win32.cxx: patch does not apply
  error: patch failed: src/drivers/Android/Fl_Android_System_Driver.H:25
  error: src/drivers/Android/Fl_Android_System_Driver.H: patch does not apply


  gmake: warning: -j16 forced in submake: resetting jobserver mode.
  In file included from /usr/x86_64-w64-mingw32/sys-root/mingw/include/windows.h:70,
                   from /home/ethan/.cargo/registry/src/github.com-1ecc6299db9ec823/fltk-sys-1.2.20/cfltk/fltk/fluid/ExternalCodeEditor_WIN32.h:25,
                   from /home/ethan/.cargo/registry/src/github.com-1ecc6299db9ec823/fltk-sys-1.2.20/cfltk/fltk/fluid/ExternalCodeEditor_WIN32.cxx:24:
  /home/ethan/.cargo/registry/src/github.com-1ecc6299db9ec823/fltk-sys-1.2.20/cfltk/fltk/fluid/ExternalCodeEditor_WIN32.cxx: In member function 'int ExternalCodeEditor::reap_editor(DWORD*)':
  /home/ethan/.cargo/registry/src/github.com-1ecc6299db9ec823/fltk-sys-1.2.20/cfltk/fltk/fluid/ExternalCodeEditor_WIN32.cxx:509:10: error: narrowing conversion of '4294967295' from 'DWORD' {aka 'long unsigned int'} to 'int' [-Wnarrowing]
    509 |     case WAIT_FAILED: {    // failed
        |          ^~~~~~~~~~~
  gmake[2]: *** [fltk/fluid/CMakeFiles/fluid.dir/build.make:422: fltk/fluid/CMakeFiles/fluid.dir/ExternalCodeEditor_WIN32.cxx.obj] Error 1
  gmake[1]: *** [CMakeFiles/Makefile2:408: fltk/fluid/CMakeFiles/fluid.dir/all] Error 2
  gmake: *** [Makefile:136: all] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2

  build script failed, must exit now', /home/ethan/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.46/src/lib.rs:974:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
And the toolchain list:
```
$ rustup toolchain list
stable-x86_64-unknown-linux-gnu (default)
nightly-2020-04-10-x86_64-unknown-linux-gnu
nightly-x86_64-unknown-linux-gnu
1.56.0-x86_64-unknown-linux-gnu
```

## Desktop info
 - OS: openSUSE Tumbleweed
 - Version: snapshot 20221214

**Comment by EthanWeegee** [comment]
Building with the last fltk version that worked ([the lockfile](https://github.com/EthanWeegee/simple-nus3audio-gui/blob/bb3c0e0395c895810b783b9fc2d533d03317a4d5/Cargo.lock#L371) says 1.2.20) also fails with the same error.

**Comment by MoAlyousef** [comment]
Hi
The log you posted seems to still be building version 1.2.20. The narrowing conversion error was already fixed upstream here:
https://github.com/fltk/fltk/commit/793f4b90fac349b096922a6b90ae2731777ac6cf

That should correspond to fltk-rs version 1.2.29 and later. 

Can you run `cargo update` to update the lockfile then rerun the build?

**Comment by EthanWeegee** [comment]
> Hi The log you posted seems to still be building version 1.2.20. The narrowing conversion error was already fixed upstream here: [fltk/fltk@793f4b9](https://github.com/fltk/fltk/commit/793f4b90fac349b096922a6b90ae2731777ac6cf)
> 
> That should correspond to fltk-rs version 1.2.29 and later.
> 
> Can you run `cargo update` to update the lockfile then rerun the build?

Thanks for the quick reply. I'm not sure how I missed that, I assumed cargo would just pull the latest version as it came up.

The error is gone and the build succeeds now.
